specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -207,17 +212,47 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
             }
           );
         }
+        // Flag as invalid if no determinations are checked as current
         if (
           determination.collection !== undefined &&
           !determination.collection.models.some(
             (c: SpecifyResource<Determination>) => c.get('isCurrent')
           )
         ) {
-          determination.set('isCurrent', true);
+          return {
+            isValid: false,
+            reason: resourcesText.currentDeterminationRequired(),
+            saveBlockerKey: CURRENT_DETERMINATION_KEY,
+            resource: determination.collection.related,
+          };
         }
-        return { isValid: true };
+        return {
+          isValid: true,
+          saveBlockerKey: CURRENT_DETERMINATION_KEY,
+          resource: determination.collection?.related,
+        };
       },
     },
+    onRemoved: (determination: SpecifyResource<Determination>): void => {
+      // Block save when a current determination is deleted
+      if (determination.get('isCurrent'))
+        setSaveBlockers(
+          determination.collection?.related ?? determination,
```

Comment:
(optional)
In this case, we know that `determination.collection` will always be defined. 
We can directly access the Collection which the resource was added to by using the second parameter of the `onRemoved` function: 
https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts#L31-L34

We trigger this `onRemoved` function whenever the `removed` Backbone event is triggered: 
https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts#L59-L60

Backbone handles passing the resource instance and the collection as arguments with the event. 
See the Backbone docs: https://backbonejs.org/#Events-catalog

```suggestion
    onRemoved: (determination, collection): void => {
      // Block save when a current determination is deleted
      if (determination.get('isCurrent'))
        setSaveBlockers(
          collection.related ?? determination,
```